Source/WebKit/WebProcess/Automation/WebAutomationSessionProxy.cpp

```diff
@@ -645,16 +645,30 @@ void WebAutomationSessionProxy::focusFrame(WebCore::PageIdentifier pageID, WebCo
         return;
     }
 
-    // If frame is no longer connected to the page, then it is
-    // closing and it's not possible to focus the frame.
-    RefPtr frame = WebProcess::singleton().webFrame(frameID);
-    if (!frame || !frame->coreFrame() || !frame->coreFrame()->page()) {
+    auto executeHandlerWithFrameNotFoundError = [&] {
```

Comment:
nit: just capture `completionHandler`

```suggestion
    auto executeHandlerWithFrameNotFoundError = [&completionHandler] {
```